make_version.py

```diff
@@ -7,7 +7,15 @@ def main():
     if len(output) == 3:
         version, build, commit = output
     else:
-        raise Exception("Could not git describe, (got %s)" % output)
+        # If the clone is shallow, describe output won't have tag and
+        # number of commits.  This is a particular issue on Travis-CI,
+        # which by default clones with a depth of 50.
+        # This behaviour isn't well documented in git-describe docs,
+        # but see, e.g., https://stackoverflow.com/a/36389573/1008142
+        # and https://github.com/travis-ci/travis-ci/issues/3412
+        version = 'unknown'
+        build = 'unknown'
+        commit = output
```

Comment:
I think that your updated changes are good.

Regarding security, the attack is still feasible with a correct `git clone` program if the repository itself is corrupted. However, like having a compromised `git` initially, this precondition is very difficult to satisfy, so I think that it is not critical to consider.

Regarding elaborate version strings, the basic argument for including the commit hash in nonrelease revisions is the existence of many users that work from repository clones rather than releases. In my experience, these users may `git clone` and then perform a system-wide installation, after which work proceeds from a different directory and the original clone directory may be deleted.

To include commit hashes or other data in the version string, [PEP 440 defines developmental releases to facilitate this](https://www.python.org/dev/peps/pep-0440/#developmental-releases), which involves ".dev", e.g., [as used in numpy](https://github.com/numpy/numpy/blob/c6533b6c386dc0f4009e5f3c5c545dde4d1b48a4/setup.py#L108-L128).

In the case of python-control, creating a conda package involves a build number. I am not aware of a definition of this value, but at least two recommendations are:

1. number of commits from most recent tag, <http://conda-test.pydata.org/docs/build.html#environment-variables-set-during-the-build-process>
2. "the number of the build" (https://docs.continuum.io/docs_oss/conda/bdist_conda#build-number), which I guess implies every time `conda build` is used, this value should increment.

However, some projects just fix this value at 0 or 1, e.g., [bison in the conda-recipes repo](https://github.com/conda/conda-recipes/blob/c2eb600f8545cd21aa9e50a8bb8a81df7fd3c915/bison/meta.yaml).